tests/lts_compatibility.py

```diff
@@ -327,15 +339,29 @@ def run_code_upgrade_from(
             LOG.info("Apply transactions to hybrid network, with primary as old node")
             issue_activity_on_live_service(network, args)
 
-            old_code_id = infra.utils.get_code_id(
+            primary, _ = network.find_primary()
+
+            old_measurement = infra.utils.get_measurement(
                 args.enclave_type,
                 args.enclave_platform,
-                args.oe_binary,
                 args.package,
                 library_dir=from_library_dir,
             )
-            primary, _ = network.find_primary()
-            network.consortium.retire_code(primary, old_code_id)
+            if old_measurement != new_measurement:
+                network.consortium.remove_measurement(
+                    primary, args.enclave_platform, old_measurement
+                )
+
+            # If host_data was found for original nodes, check if it's different on new nodes, in which case old should be removed
+            if new_host_data is not None:
+                old_host_data, old_security_policy = (
+                    infra.utils.get_host_data_and_security_policy(args.enclave_platform)
+                )
+
+                if old_host_data != new_host_data:
+                    network.consortium.remove_host_data(
+                        primary, args.enclave_platform, old_host_data
+                    )
```

Comment:
This is hopefully predicting what will be necessary if/when a version-transition includes a host-data update. Currently it doesn't, on either SNP _or_ Virtual. But if it _does_, we will (probably?) want to cycle it like we cycle measurements (I hope?).

Reply:
How come that's not used? It seems like code updates on Virtual should go through this exact path, as we code upgrade from one binary to the next.

Reply:
This path is now taken, following the changes discussed below (host_data is now the enclave digest).